util/make_new_dif.py

```diff
@@ -133,7 +150,30 @@ def main():
     args = parser.parse_args()
 
     # Parse CMD line args.
-    ip = Ip(args.ip, args.peripheral)
+    ips = []
+
+    # Check for batch generation mode (used in CI check:
+    # ci/scripts/check-generated.sh")
+    if args.mode == "batch":
+        assert (
+            len(args.only) == 1 and args.only[0] == "autogen" and
+            "ERROR: can only batch generate completely auto-generated code.")
+        # Create list of IPs to re-generate DIF code for.
+        for autogen_src_filename in glob.iglob(
+                "sw/device/lib/dif/autogen/*.c"):
+            # NOTE: the line below takes as input a file path
+            # (/path/to/dif_uart_autogen.c) and returns the IP name in lower
+            # case snake mode (i.e., uart).
+            ip_name_snake = os.path.basename(autogen_src_filename)[4:-10]
```

Comment:
Is there something cleaner we can do here with Pathlib?

(I am of the opinion we should stop writing python, but if we do, we should at least use pathlib, which is... easier to review.)

Reply:
Fixed.